lua/ibl/init.lua

```diff
@@ -369,10 +389,67 @@ M.refresh = function(bufnr)
 
         last_whitespace_tbl = whitespace_tbl
 
-        -- #### make virtual text ####
-        local scope_start = row == scope_row_start
-        local scope_end = row == scope_row_end
-        if scope_start and scope then
+        --- current indent
+        if current_indent_enabled and row == lnum then
+            local current_indent_whitespace_tbl = vim.tbl_extend("keep", {}, whitespace_tbl)
+
+            while #current_indent_whitespace_tbl > 0 do
+                if indent.is_indent(current_indent_whitespace_tbl[#current_indent_whitespace_tbl]) then
+                    table.remove(current_indent_whitespace_tbl, #current_indent_whitespace_tbl)
+                    break
+                end
+                table.remove(current_indent_whitespace_tbl, #current_indent_whitespace_tbl)
+            end
+
+            current_indent_col_start_single = #current_indent_whitespace_tbl
+
+            for j = indent_state and #indent_state.stack or 0, 1, -1 do
+                if indent_state.stack[j].indent < current_indent_col_start_single then
+                    break
+                end
+                current_indent.start_row = indent_state.stack[j].row + 1
+            end
+            if current_indent.start_row > row then
+                current_indent.start_row = math.huge
+            end
+
+            current_indent_index = #vim.tbl_filter(function(w)
+                return indent.is_indent(w)
+            end, current_indent_whitespace_tbl) + 1
+        end
+
+        do
+            calc_data[i] = { whitespace, whitespace_tbl }
+        end
+
+        ::continue_calc::
+    end
+
+    for _, fn in
+        pairs(hooks.get(bufnr, hooks.type.CURRENT_INDENT_HIGHLIGHT) --[[ @as ibl.hooks.cb.current_indent_highlight[] ]])
+    do
+        current_indent_index = fn(buffer_state.tick, bufnr, current_indent, current_indent_index)
+    end
+
+    --- Draw loop ---
+    for i, line in ipairs(lines) do
+        local row = i + offset
+        if line_skipped[i] then
+            vt.clear_buffer(bufnr, row)
+            goto continue_draw
+        end
+        local whitespace, whitespace_tbl = unpack(calc_data[i])
+
+        local is_current_indent_active = row >= current_indent.start_row
+            and row <= (current_indent.end_row or #lines + offset)
+        local is_current_indent_start = row + 1 == (current_indent.start_row or -1)
+        local is_current_indent_end = row == (current_indent.end_row or -1)
```

Comment:
I think `current_indent.start_row` and `current_indent.end_row` are always set, so I am unsure what the purpose of the `or`s are here? And if you want to use `or` to cover all cases, the first `start_row` should also have an `or -1`.